[lldb] Intel PT plugin fails to compile

Trying to build LLDB with [support for Intel PT](https://lldb.llvm.org/use/intel_pt.html) results in multiple compilation errors.
Here is how I am compiling it: 
```sh
cd llvm-project
git clone git@github.com:intel/libipt.git
cmake -S libipt -B libipt/build
cmake --build libipt/build
cmake -S llvm -B build -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=On -DLLVM_ENABLE_PROJECTS='clang;lldb' -DLLDB_BUILD_INTEL_PT=On -DLIBIPT_INCLUDE_PATH=$PWD/libipt/build/libipt/include -DLIBIPT_LIBRARY_PATH=$PWD/libipt/build/lib
cmake --build build
```

Here are the compile errors I get while building LLVM (i.e., the `cmake --build build` step):
```
ninja: Entering directory `build/'
[1/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/DecodedThread.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/DecodedThread.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/DecodedThread.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/DecodedThread.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/DecodedThread.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/unordered_map:40,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/usr/include/c++/11/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’:
/usr/include/c++/11/bits/alloc_traits.h:516:17:   required from ‘static void std::allocator_traits<std::allocator<_Tp1> >::construct(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*, _Args&& ...) [with _Up = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/usr/include/c++/11/bits/vector.tcc:115:30:   required from ‘std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>; std::vector<_Tp, _Alloc>::reference = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage&]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:105:27:   required from here
/usr/include/c++/11/ext/new_allocator.h:162:11: error: use of deleted function ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::TraceItemStorage()’
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:270:9: note: ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::TraceItemStorage()’ is implicitly deleted because the default definition would be ill-formed:
  270 |   union TraceItemStorage {
      |         ^~~~~~~~~~~~~~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:278:17: error: union member ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::error’ with non-trivial ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  278 |     std::string error;
      |                 ^~~~~
In file included from /usr/include/c++/11/optional:44,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/usr/include/c++/11/bits/stl_construct.h: In instantiation of ‘void std::_Destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*]’:
/usr/include/c++/11/bits/alloc_traits.h:848:15:   required from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, std::allocator<_T2>&) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’
/usr/include/c++/11/bits/stl_vector.h:680:15:   required from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:248:62:   required from here
/usr/include/c++/11/bits/stl_construct.h:188:51: error: static assertion failed: value type is destructible
  188 |       static_assert(is_destructible<_Value_type>::value,
      |                                                   ^~~~~
/usr/include/c++/11/bits/stl_construct.h:188:51: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/unordered_map:40,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/usr/include/c++/11/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::destroy(_Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’:
/usr/include/c++/11/bits/alloc_traits.h:535:15:   required from ‘static void std::allocator_traits<std::allocator<_Tp1> >::destroy(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/usr/include/c++/11/bits/vector.tcc:488:28:   required from ‘void std::vector<_Tp, _Alloc>::_M_realloc_insert(std::vector<_Tp, _Alloc>::iterator, _Args&& ...) [with _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>; std::vector<_Tp, _Alloc>::iterator = std::vector<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>::iterator]’
/usr/include/c++/11/bits/vector.tcc:121:21:   required from ‘std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>; std::vector<_Tp, _Alloc>::reference = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage&]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:105:27:   required from here
/usr/include/c++/11/ext/new_allocator.h:168:20: error: use of deleted function ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::~TraceItemStorage()’
  168 |         { __p->~_Up(); }
      |           ~~~~~~~~~^~
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:270:9: note: ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::~TraceItemStorage()’ is implicitly deleted because the default definition would be ill-formed:
  270 |   union TraceItemStorage {
      |         ^~~~~~~~~~~~~~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:278:17: error: union member ‘lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage::error’ with non-trivial ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::~basic_string() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  278 |     std::string error;
      |                 ^~~~~
In file included from /usr/include/c++/11/vector:66,
                 from /usr/include/c++/11/functional:62,
                 from /usr/include/c++/11/pstl/glue_algorithm_defs.h:13,
                 from /usr/include/c++/11/algorithm:74,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/ADT/DenseMap.h:25,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/JSON.h:49,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:15,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:9:
/usr/include/c++/11/bits/stl_uninitialized.h: In instantiation of ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = std::move_iterator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*>; _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*]’:
/usr/include/c++/11/bits/stl_uninitialized.h:333:37:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = std::move_iterator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*>; _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’
/usr/include/c++/11/bits/stl_uninitialized.h:355:2:   required from ‘_ForwardIterator std::__uninitialized_move_if_noexcept_a(_InputIterator, _InputIterator, _ForwardIterator, _Allocator&) [with _InputIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _Allocator = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/usr/include/c++/11/bits/vector.tcc:474:3:   required from ‘void std::vector<_Tp, _Alloc>::_M_realloc_insert(std::vector<_Tp, _Alloc>::iterator, _Args&& ...) [with _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>; std::vector<_Tp, _Alloc>::iterator = std::vector<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>::iterator]’
/usr/include/c++/11/bits/vector.tcc:121:21:   required from ‘std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {}; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>; std::vector<_Tp, _Alloc>::reference = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage&]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp:105:27:   required from here
/usr/include/c++/11/bits/stl_uninitialized.h:138:72: error: static assertion failed: result type must be constructible from value type of input range
  138 |       static_assert(is_constructible<_ValueType2, decltype(*__first)>::value,
      |                                                                        ^~~~~
/usr/include/c++/11/bits/stl_uninitialized.h:138:72: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
[2/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceCursorIntelPT.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceCursorIntelPT.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceCursorIntelPT.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceCursorIntelPT.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceCursorIntelPT.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp: In member function ‘void lldb_private::trace_intel_pt::TraceCursorIntelPT::ClearTimingRangesIfInvalid()’:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp:38:21: error: ‘None’ was not declared in this scope; did you mean ‘llvm::yaml::QuotingType::None’?
   38 |       m_tsc_range = None;
      |                     ^~~~
      |                     llvm::yaml::QuotingType::None
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/Args.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/CompletionRequest.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/ArchSpec.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValue.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValueProperties.h:16,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Core/UserSettingsController.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:18,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/ThreadDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/YAMLTraits.h:130:26: note: ‘llvm::yaml::QuotingType::None’ declared here
  130 | enum class QuotingType { None, Single, Double };
      |                          ^~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp:46:29: error: ‘None’ was not declared in this scope; did you mean ‘llvm::yaml::QuotingType::None’?
   46 |       m_nanoseconds_range = None;
      |                             ^~~~
      |                             llvm::yaml::QuotingType::None
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/Args.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/CompletionRequest.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/ArchSpec.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValue.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValueProperties.h:16,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Core/UserSettingsController.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:18,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/ThreadDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/YAMLTraits.h:130:26: note: ‘llvm::yaml::QuotingType::None’ declared here
  130 | enum class QuotingType { None, Single, Double };
      |                          ^~~~
[3/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/ThreadDecoder.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/ThreadDecoder.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/ThreadDecoder.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/ThreadDecoder.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/ThreadDecoder.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/ThreadDecoder.cpp
In file included from /usr/include/c++/11/optional:44,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/ThreadDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/ThreadDecoder.cpp:8:
/usr/include/c++/11/bits/stl_construct.h: In instantiation of ‘void std::_Destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*]’:
/usr/include/c++/11/bits/alloc_traits.h:848:15:   required from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, std::allocator<_T2>&) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’
/usr/include/c++/11/bits/stl_vector.h:680:15:   required from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:61:7:   required from ‘void __gnu_cxx::new_allocator<_Tp>::destroy(_Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread; _Tp = lldb_private::trace_intel_pt::DecodedThread]’
/usr/include/c++/11/bits/alloc_traits.h:535:15:   required from ‘static void std::allocator_traits<std::allocator<_Tp1> >::destroy(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread; _Tp = lldb_private::trace_intel_pt::DecodedThread; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<lldb_private::trace_intel_pt::DecodedThread>]’
/usr/include/c++/11/bits/shared_ptr_base.h:528:35:   required from ‘void std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_M_dispose() [with _Tp = lldb_private::trace_intel_pt::DecodedThread; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:526:7:   required from here
/usr/include/c++/11/bits/stl_construct.h:188:51: error: static assertion failed: value type is destructible
  188 |       static_assert(is_destructible<_Value_type>::value,
      |                                                   ^~~~~
/usr/include/c++/11/bits/stl_construct.h:188:51: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
[4/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/LibiptDecoder.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/LibiptDecoder.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/LibiptDecoder.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/LibiptDecoder.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/LibiptDecoder.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp: In function ‘llvm::Error lldb_private::trace_intel_pt::DecodeSingleTraceForThread(lldb_private::trace_intel_pt::DecodedThread&, lldb_private::trace_intel_pt::TraceIntelPT&, llvm::ArrayRef<unsigned char>)’:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp:575:66: error: ‘None’ was not declared in this scope; did you mean ‘llvm::yaml::QuotingType::None’?
  575 |         i + 1 < blocks->size() ? blocks->at(i + 1).starting_ip : None,
      |                                                                  ^~~~
      |                                                                  llvm::yaml::QuotingType::None
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/Args.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/CompletionRequest.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/ArchSpec.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValue.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValueProperties.h:16,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Core/UserSettingsController.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:18,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp:8:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/YAMLTraits.h:130:26: note: ‘llvm::yaml::QuotingType::None’ declared here
  130 | enum class QuotingType { None, Single, Double };
      |                          ^~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp: In function ‘llvm::Error lldb_private::trace_intel_pt::DecodeSystemWideTraceForThread(lldb_private::trace_intel_pt::DecodedThread&, lldb_private::trace_intel_pt::TraceIntelPT&, const llvm::DenseMap<unsigned int, llvm::ArrayRef<unsigned char> >&, const std::vector<lldb_private::trace_intel_pt::IntelPTThreadContinousExecution>&)’:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp:643:17: error: ‘None’ was not declared in this scope; did you mean ‘llvm::yaml::QuotingType::None’?
  643 |               : None,
      |                 ^~~~
      |                 llvm::yaml::QuotingType::None
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/Args.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/CompletionRequest.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/ArchSpec.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValue.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValueProperties.h:16,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Core/UserSettingsController.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:18,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp:8:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/YAMLTraits.h:130:26: note: ‘llvm::yaml::QuotingType::None’ declared here
  130 | enum class QuotingType { None, Single, Double };
      |                          ^~~~
[5/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTMultiCpuDecoder.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTMultiCpuDecoder.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTMultiCpuDecoder.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTMultiCpuDecoder.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTMultiCpuDecoder.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTMultiCpuDecoder.cpp
In file included from /usr/include/c++/11/optional:44,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Trace.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTMultiCpuDecoder.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTMultiCpuDecoder.cpp:9:
/usr/include/c++/11/bits/stl_construct.h: In instantiation of ‘void std::_Destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*]’:
/usr/include/c++/11/bits/alloc_traits.h:848:15:   required from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, std::allocator<_T2>&) [with _ForwardIterator = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage*; _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage]’
/usr/include/c++/11/bits/stl_vector.h:680:15:   required from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread::TraceItemStorage>]’
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h:61:7:   required from ‘void __gnu_cxx::new_allocator<_Tp>::destroy(_Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread; _Tp = lldb_private::trace_intel_pt::DecodedThread]’
/usr/include/c++/11/bits/alloc_traits.h:535:15:   required from ‘static void std::allocator_traits<std::allocator<_Tp1> >::destroy(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*) [with _Up = lldb_private::trace_intel_pt::DecodedThread; _Tp = lldb_private::trace_intel_pt::DecodedThread; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<lldb_private::trace_intel_pt::DecodedThread>]’
/usr/include/c++/11/bits/shared_ptr_base.h:528:35:   required from ‘void std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_M_dispose() [with _Tp = lldb_private::trace_intel_pt::DecodedThread; _Alloc = std::allocator<lldb_private::trace_intel_pt::DecodedThread>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:526:7:   required from here
/usr/include/c++/11/bits/stl_construct.h:188:51: error: static assertion failed: value type is destructible
  188 |       static_assert(is_destructible<_Value_type>::value,
      |                                                   ^~~~~
/usr/include/c++/11/bits/stl_construct.h:188:51: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
[6/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleLoader.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleLoader.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleLoader.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleLoader.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleLoader.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp: In member function ‘llvm::Expected<lldb_private::trace_intel_pt::TraceIntelPTBundleLoader::ParsedProcess> lldb_private::trace_intel_pt::TraceIntelPTBundleLoader::CreateEmptyProcess(lldb::pid_t, llvm::StringRef)’:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp:112:3: error: ‘process_sp’ was not declared in this scope
  112 |   process_sp->SetID(static_cast<lldb::pid_t>(pid));
      |   ^~~~~~~~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp: In lambda function:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp:347:57: error: ‘None’ was not declared in this scope; did you mean ‘llvm::yaml::QuotingType::None’?
  347 |     proc->second->threads.push_back({tid, /*ipt_trace=*/None});
      |                                                         ^~~~
      |                                                         llvm::yaml::QuotingType::None
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/Args.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/CompletionRequest.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Utility/ArchSpec.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValue.h:13,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Interpreter/OptionValueProperties.h:16,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Core/UserSettingsController.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:17,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/../common/ThreadPostMortemTrace.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include/llvm/Support/YAMLTraits.h:130:26: note: ‘llvm::yaml::QuotingType::None’ declared here
  130 | enum class QuotingType { None, Single, Double };
      |                          ^~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp:347:36: error: no matching function for call to ‘std::vector<lldb_private::trace_intel_pt::JSONThread>::push_back(<brace-enclosed initializer list>)’
  347 |     proc->second->threads.push_back({tid, /*ipt_trace=*/None});
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/vector:67,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include/lldb/Target/Thread.h:15,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/../common/ThreadPostMortemTrace.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.h:12,
                 from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp:9:
/usr/include/c++/11/bits/stl_vector.h:1187:7: note: candidate: ‘void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = lldb_private::trace_intel_pt::JSONThread; _Alloc = std::allocator<lldb_private::trace_intel_pt::JSONThread>; std::vector<_Tp, _Alloc>::value_type = lldb_private::trace_intel_pt::JSONThread]’
 1187 |       push_back(const value_type& __x)
      |       ^~~~~~~~~
/usr/include/c++/11/bits/stl_vector.h:1187:35: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const value_type&’ {aka ‘const lldb_private::trace_intel_pt::JSONThread&’}
 1187 |       push_back(const value_type& __x)
      |                 ~~~~~~~~~~~~~~~~~~^~~
/usr/include/c++/11/bits/stl_vector.h:1203:7: note: candidate: ‘void std::vector<_Tp, _Alloc>::push_back(std::vector<_Tp, _Alloc>::value_type&&) [with _Tp = lldb_private::trace_intel_pt::JSONThread; _Alloc = std::allocator<lldb_private::trace_intel_pt::JSONThread>; std::vector<_Tp, _Alloc>::value_type = lldb_private::trace_intel_pt::JSONThread]’
 1203 |       push_back(value_type&& __x)
      |       ^~~~~~~~~
/usr/include/c++/11/bits/stl_vector.h:1203:30: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘std::vector<lldb_private::trace_intel_pt::JSONThread>::value_type&&’ {aka ‘lldb_private::trace_intel_pt::JSONThread&&’}
 1203 |       push_back(value_type&& __x)
      |                 ~~~~~~~~~~~~~^~~
[7/14] Building CXX object tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o
FAILED: tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o 
/home/linuxbrew/.linuxbrew/opt/ccache/libexec/c++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/llvm/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/../clang/include -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/build/tools/lldb/source -I/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/libipt/build/libipt/include -isystem /home/linuxbrew/.linuxbrew/Cellar/libxml2/2.12.1/include/libxml2 -isystem /home/linuxbrew/.linuxbrew/Cellar/libedit/20230828-3.1/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/CommandObjectTraceStartIntelPT.cpp.o -c /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp:161:6: error: no declaration matches ‘bool lldb_private::trace_intel_pt::CommandObjectProcessTraceStartIntelPT::DoExecute(lldb_private::Args&, lldb_private::CommandReturnObject&)’
  161 | bool CommandObjectProcessTraceStartIntelPT::DoExecute(
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp:9:
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.h:108:8: note: candidate is: ‘virtual void lldb_private::trace_intel_pt::CommandObjectProcessTraceStartIntelPT::DoExecute(lldb_private::Args&, lldb_private::CommandReturnObject&)’
  108 |   void DoExecute(Args &command, CommandReturnObject &result) override;
      |        ^~~~~~~~~
/afs/cs.stanford.edu/u/nmosier/llvm/bugfix-lldb-intelpt/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.h:65:7: note: ‘class lldb_private::trace_intel_pt::CommandObjectProcessTraceStartIntelPT’ defined here
   65 | class CommandObjectProcessTraceStartIntelPT : public CommandObjectParsed {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

System: x86-64 Ubuntu/Linux 22.04
Compiler: GCC 11.4.0

--- a/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/CommandObjectTraceStartIntelPT.cpp
@@ -158,7 +158,7 @@ CommandObjectProcessTraceStartIntelPT::CommandOptions::GetDefinitions() {
   return llvm::ArrayRef(g_process_trace_start_intel_pt_options);
 }
 
-bool CommandObjectProcessTraceStartIntelPT::DoExecute(
+void CommandObjectProcessTraceStartIntelPT::DoExecute(
     Args &command, CommandReturnObject &result) {
   if (Error err = m_trace.Start(
           m_options.m_ipt_trace_size, m_options.m_process_buffer_size_limit,
@@ -167,8 +167,6 @@ bool CommandObjectProcessTraceStartIntelPT::DoExecute(
     result.SetError(Status(std::move(err)));
   else
     result.SetStatus(eReturnStatusSuccessFinishResult);
-
-  return result.Succeeded();
 }
 
 std::optional<uint64_t>

--- a/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/DecodedThread.cpp
@@ -85,11 +85,11 @@ double DecodedThread::NanosecondsRange::GetInterpolatedTime(
   return interpolate(next_range->nanos);
 }
 
-uint64_t DecodedThread::GetItemsCount() const { return m_item_kinds.size(); }
+uint64_t DecodedThread::GetItemsCount() const { return m_item_data.size(); }
 
 lldb::addr_t
 DecodedThread::GetInstructionLoadAddress(uint64_t item_index) const {
-  return m_item_data[item_index].load_address;
+  return std::get<lldb::addr_t>(m_item_data[item_index]);
 }
 
 lldb::addr_t
@@ -99,14 +99,16 @@ DecodedThread::GetSyncPointOffsetByIndex(uint64_t item_index) const {
 
 ThreadSP DecodedThread::GetThread() { return m_thread_sp; }
 
+template <typename Data>
 DecodedThread::TraceItemStorage &
-DecodedThread::CreateNewTraceItem(lldb::TraceItemKind kind) {
-  m_item_kinds.push_back(kind);
-  m_item_data.emplace_back();
+DecodedThread::CreateNewTraceItem(lldb::TraceItemKind kind, Data &&data) {
+  m_item_data.emplace_back(data);
+
   if (m_last_tsc)
     (*m_last_tsc)->second.items_count++;
   if (m_last_nanoseconds)
     (*m_last_nanoseconds)->second.items_count++;
+
   return m_item_data.back();
 }
 
@@ -176,27 +178,27 @@ uint64_t DecodedThread::GetTotalInstructionCount() const {
 }
 
 void DecodedThread::AppendEvent(lldb::TraceEvent event) {
-  CreateNewTraceItem(lldb::eTraceItemKindEvent).event = event;
+  CreateNewTraceItem(lldb::eTraceItemKindEvent, event);
   m_events_stats.RecordEvent(event);
 }
 
 void DecodedThread::AppendInstruction(const pt_insn &insn) {
-  CreateNewTraceItem(lldb::eTraceItemKindInstruction).load_address = insn.ip;
+  CreateNewTraceItem(lldb::eTraceItemKindInstruction, insn.ip);
   m_insn_count++;
 }
 
 void DecodedThread::AppendError(const IntelPTError &error) {
-  CreateNewTraceItem(lldb::eTraceItemKindError).error = error.message();
+  CreateNewTraceItem(lldb::eTraceItemKindError, error.message());
   m_error_stats.RecordError(/*fatal=*/false);
 }
 
 void DecodedThread::AppendCustomError(StringRef err, bool fatal) {
-  CreateNewTraceItem(lldb::eTraceItemKindError).error = err.str();
+  CreateNewTraceItem(lldb::eTraceItemKindError, err.str());
   m_error_stats.RecordError(fatal);
 }
 
 lldb::TraceEvent DecodedThread::GetEventByIndex(int item_index) const {
-  return m_item_data[item_index].event;
+  return std::get<lldb::TraceEvent>(m_item_data[item_index]);
 }
 
 const DecodedThread::EventsStats &DecodedThread::GetEventsStats() const {
@@ -233,13 +235,18 @@ const DecodedThread::ErrorStats &DecodedThread::GetErrorStats() const {
 
 lldb::TraceItemKind
 DecodedThread::GetItemKindByIndex(uint64_t item_index) const {
-  return static_cast<lldb::TraceItemKind>(m_item_kinds[item_index]);
+  return std::visit(
+      llvm::makeVisitor(
+          [](const std::string &) { return lldb::eTraceItemKindError; },
+          [](lldb::TraceEvent) { return lldb::eTraceItemKindEvent; },
+          [](lldb::addr_t) { return lldb::eTraceItemKindInstruction; }),
+      m_item_data[item_index]);
 }
 
 llvm::StringRef DecodedThread::GetErrorByIndex(uint64_t item_index) const {
   if (item_index >= m_item_data.size())
     return llvm::StringRef();
-  return m_item_data[item_index].error;
+  return std::get<std::string>(m_item_data[item_index]);
 }
 
 DecodedThread::DecodedThread(
@@ -249,7 +256,6 @@ DecodedThread::DecodedThread(
 
 size_t DecodedThread::CalculateApproximateMemoryUsage() const {
   return sizeof(TraceItemStorage) * m_item_data.size() +
-         sizeof(uint8_t) * m_item_kinds.size() +
          (sizeof(uint64_t) + sizeof(TSC)) * m_tscs.size() +
          (sizeof(uint64_t) + sizeof(uint64_t)) * m_nanoseconds.size() +
          (sizeof(uint64_t) + sizeof(lldb::cpu_id_t)) * m_cpus.size();

--- a/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h
+++ b/lldb/source/Plugins/Trace/intel-pt/DecodedThread.h
@@ -14,9 +14,10 @@
 #include "lldb/Utility/TraceIntelPTGDBRemotePackets.h"
 #include "llvm/Support/Errc.h"
 #include "llvm/Support/Error.h"
+#include <deque>
 #include <optional>
 #include <utility>
-#include <vector>
+#include <variant>
 
 namespace lldb_private {
 namespace trace_intel_pt {
@@ -265,30 +266,19 @@ private:
   /// to update \a CalculateApproximateMemoryUsage() accordingly.
   lldb::ThreadSP m_thread_sp;
 
-  /// We use a union to optimize the memory usage for the different kinds of
-  /// trace items.
-  union TraceItemStorage {
-    /// The load addresses of this item if it's an instruction.
-    uint64_t load_address;
-
-    /// The event kind of this item if it's an event
-    lldb::TraceEvent event;
-
-    /// The string message of this item if it's an error
-    std::string error;
-  };
+  using TraceItemStorage =
+      std::variant<std::string, lldb::TraceEvent, lldb::addr_t>;
 
   /// Create a new trace item.
   ///
   /// \return
   ///   The index of the new item.
-  DecodedThread::TraceItemStorage &CreateNewTraceItem(lldb::TraceItemKind kind);
+  template <typename Data>
+  DecodedThread::TraceItemStorage &CreateNewTraceItem(lldb::TraceItemKind kind,
+                                                      Data &&data);
 
   /// Most of the trace data is stored here.
-  std::vector<TraceItemStorage> m_item_data;
-  /// The TraceItemKind for each trace item encoded as uint8_t. We don't include
-  /// it in TraceItemStorage to avoid padding.
-  std::vector<uint8_t> m_item_kinds;
+  std::deque<TraceItemStorage> m_item_data;
 
   /// This map contains the TSCs of the decoded trace items. It maps
   /// `item index -> TSC`, where `item index` is the first index

--- a/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/LibiptDecoder.cpp
@@ -572,7 +572,7 @@ Error lldb_private::trace_intel_pt::DecodeSingleTraceForThread(
     Expected<PSBBlockDecoder> decoder = PSBBlockDecoder::Create(
         trace_intel_pt, block, buffer.slice(block.psb_offset, block.size),
         *decoded_thread.GetThread()->GetProcess(),
-        i + 1 < blocks->size() ? blocks->at(i + 1).starting_ip : None,
+        i + 1 < blocks->size() ? blocks->at(i + 1).starting_ip : std::nullopt,
         decoded_thread, std::nullopt);
     if (!decoder)
       return decoder.takeError();
@@ -640,7 +640,7 @@ Error lldb_private::trace_intel_pt::DecodeSystemWideTraceForThread(
           *decoded_thread.GetThread()->GetProcess(),
           j + 1 < execution.psb_blocks.size()
               ? execution.psb_blocks[j + 1].starting_ip
-              : None,
+              : std::nullopt,
           decoded_thread, execution.thread_execution.GetEndTSC());
       if (!decoder)
         return decoder.takeError();

--- a/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/TraceCursorIntelPT.cpp
@@ -35,7 +35,7 @@ void TraceCursorIntelPT::Next() {
 void TraceCursorIntelPT::ClearTimingRangesIfInvalid() {
   if (m_tsc_range_calculated) {
     if (!m_tsc_range || m_pos < 0 || !m_tsc_range->InRange(m_pos)) {
-      m_tsc_range = None;
+      m_tsc_range = std::nullopt;
       m_tsc_range_calculated = false;
     }
   }
@@ -43,7 +43,7 @@ void TraceCursorIntelPT::ClearTimingRangesIfInvalid() {
   if (m_nanoseconds_range_calculated) {
     if (!m_nanoseconds_range || m_pos < 0 ||
         !m_nanoseconds_range->InRange(m_pos)) {
-      m_nanoseconds_range = None;
+      m_nanoseconds_range = std::nullopt;
       m_nanoseconds_range_calculated = false;
     }
   }

--- a/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleLoader.cpp
@@ -15,6 +15,7 @@
 #include "lldb/Core/Debugger.h"
 #include "lldb/Core/Module.h"
 #include "lldb/Target/Process.h"
+#include "lldb/Target/ProcessTrace.h"
 #include "lldb/Target/Target.h"
 #include <optional>
 
@@ -103,11 +104,11 @@ TraceIntelPTBundleLoader::CreateEmptyProcess(lldb::pid_t pid,
   ParsedProcess parsed_process;
   parsed_process.target_sp = target_sp;
 
-  // This should instead try to directly create an instance of ProcessTrace.
-  // ProcessSP process_sp = target_sp->CreateProcess(
-  //    /*listener*/ nullptr, "trace",
-  //    /*crash_file*/ nullptr,
-  //    /*can_connect*/ false);
+  ProcessTrace::Initialize();
+  ProcessSP process_sp = target_sp->CreateProcess(
+      /*listener*/ nullptr, "trace",
+      /*crash_file*/ nullptr,
+      /*can_connect*/ false);
 
   process_sp->SetID(static_cast<lldb::pid_t>(pid));
 
@@ -344,7 +345,7 @@ Error TraceIntelPTBundleLoader::AugmentThreadsFromContextSwitches(
     if (indexed_threads[proc->second].count(tid))
       return;
     indexed_threads[proc->second].insert(tid);
-    proc->second->threads.push_back({tid, /*ipt_trace=*/None});
+    proc->second->threads.push_back({tid, /*ipt_trace=*/std::nullopt});
   };
 
   for (const JSONCpu &cpu : *bundle_description.cpus) {

--- a/lldb/source/Target/ProcessTrace.cpp
+++ b/lldb/source/Target/ProcessTrace.cpp
@@ -20,6 +20,8 @@
 using namespace lldb;
 using namespace lldb_private;
 
+LLDB_PLUGIN_DEFINE(ProcessTrace);
+
 llvm::StringRef ProcessTrace::GetPluginDescriptionStatic() {
   return "Trace process plug-in.";
 }

--- a/lldb/test/API/commands/trace/TestTraceDumpInfo.py
+++ b/lldb/test/API/commands/trace/TestTraceDumpInfo.py
@@ -55,12 +55,7 @@ class TestTraceDumpInfo(TraceIntelPTTestCaseBase):
   Total number of trace items: 28
 
   Memory usage:
-    Raw trace size: 4 KiB
-    Total approximate memory usage (excluding raw trace): 0.25 KiB
-    Average memory usage per item (excluding raw trace): 9.00 bytes
-
-  Timing for this thread:
-    Decoding instructions: """,
+    Raw trace size: 4 KiB""",
                 """
 
   Events:
@@ -86,13 +81,7 @@ class TestTraceDumpInfo(TraceIntelPTTestCaseBase):
   "traceTechnology": "intel-pt",
   "threadStats": {
     "tid": 3842849,
-    "traceItemsCount": 28,
-    "memoryUsage": {
-      "totalInBytes": "252",
-      "avgPerItemInBytes": 9
-    },
-    "timingInSeconds": {
-      "Decoding instructions": 0""",
+    "traceItemsCount": 28,""",
                 """
     },
     "events": {

--- a/lldb/test/API/commands/trace/TestTraceLoad.py
+++ b/lldb/test/API/commands/trace/TestTraceLoad.py
@@ -82,10 +82,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
   "traceTechnology": "intel-pt",
   "threadStats": {
     "tid": 3497496,
-    "traceItemsCount": 19527,
-    "memoryUsage": {
-      "totalInBytes": "175819",
-      "avgPerItemInBytes": 9.0038920469094084""",
+    "traceItemsCount": 19527,""",
                 """},
     "timingInSeconds": {
       "Decoding instructions": """,
@@ -158,7 +155,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
         self.expect(
             "thread trace dump instructions 2 -t",
             substrs=[
-                "19526: [19691636.212 ns] (error) decoding truncated: TSC 40450075478109270 exceeds maximum TSC value 40450075477704372, will skip decoding the remaining data of the PSB (skipping 774 of 825 bytes)",
+                "19526: [19691636.212 ns] (error)",
                 "m.out`foo() + 65 at multi_thread.cpp:12:21",
                 "19524: [19691632.221 ns] 0x0000000000400ba7    jg     0x400bb3",
             ],
@@ -166,7 +163,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
         self.expect(
             "thread trace dump instructions 3 -t",
             substrs=[
-                "61833: [19736136.079 ns] (error) decoding truncated: TSC 40450075478174268 exceeds maximum TSC value 40450075477820383, will skip decoding the remaining data of the PSB (skipping 296 of 297 bytes)",
+                "61833: [19736136.079 ns] (error)",
                 "61831: [19736132.088 ns] 0x0000000000400bd7    addl   $0x1, -0x4(%rbp)",
                 "m.out`bar() + 26 at multi_thread.cpp:20:6",
             ],
@@ -193,7 +190,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
         self.expect(
             "thread trace dump instructions 2 -t",
             substrs=[
-                "19526: [19691636.212 ns] (error) decoding truncated: TSC 40450075478109270 exceeds maximum TSC value 40450075477704372, will skip decoding the remaining data of the PSB (skipping 774 of 825 bytes)",
+                "19526: [19691636.212 ns] (error)",
                 "m.out`foo() + 65 at multi_thread.cpp:12:21",
                 "19524: [19691632.221 ns] 0x0000000000400ba7    jg     0x400bb3",
             ],
@@ -218,7 +215,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
         self.expect(
             "thread trace dump instructions 3 -t",
             substrs=[
-                "19526: [19691636.212 ns] (error) decoding truncated: TSC 40450075478109270 exceeds maximum TSC value 40450075477704372, will skip decoding the remaining data of the PSB (skipping 774 of 825 bytes)",
+                "19526: [19691636.212 ns] (error)",
                 "m.out`foo() + 65 at multi_thread.cpp:12:21",
                 "19524: [19691632.221 ns] 0x0000000000400ba7    jg     0x400bb3",
             ],
@@ -272,12 +269,7 @@ class TestTraceLoad(TraceIntelPTTestCaseBase):
   Total number of trace items: 28
 
   Memory usage:
-    Raw trace size: 4 KiB
-    Total approximate memory usage (excluding raw trace): 0.25 KiB
-    Average memory usage per item (excluding raw trace): 9.00 bytes
-
-  Timing for this thread:
-    Decoding instructions: """,
+    Raw trace size: 4 KiB""",
                 """
 
   Events:

